# Estimation and Confidence Intervals

In [14]:
# import all necessary packages
import numpy as np
import scipy.stats as stats
import statsmodels
import warnings 
warnings.filterwarnings('ignore')

#### Scenario
A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.


In [15]:
# Data
sample_data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
sample_data

[1.13,
 1.55,
 1.43,
 0.92,
 1.25,
 1.36,
 1.32,
 0.85,
 1.07,
 1.48,
 1.2,
 1.33,
 1.18,
 1.22,
 1.29]

In [16]:
# sample size
n=len(sample_data)
n

15

In [17]:
# sample mean x-bar (Point Estimation)
x_bar=np.mean(sample_data)
x_bar

np.float64(1.2386666666666666)

In [18]:
# Sample standard deviation
s=np.std(sample_data,ddof=1) ##\ddof=1 is for to calculate the std of sample data
s

np.float64(0.19316412956959936)

In [19]:
# degrees of freedom (n-1)
df=n-1
df

14

### Tasks

**a. Build 99% Confidence Interval Using Sample Standard Deviation**

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [20]:
# Standard Error, SE=s/sqrt(n)
SE = s / np.sqrt(n)
SE

np.float64(0.04987476379384733)

In [21]:
# 99% confidence interval -- t-distribution
co_in= stats.t.interval(0.99, df=df, loc=x_bar, scale=SE) #(confidence level,degree of freedom, mean, standard error)
co_in

(np.float64(1.0901973384384906), np.float64(1.3871359948948425))

### Final Result and Method Rationale

#### Lower and Upper Boundary Results
The 99% confidence interval provides an estimated range from **1.0902 to 1.3871 million characters**:

*   **Lower Bound (1.0902):** This value represents the minimum expected threshold for the true population mean durability.
*   **Upper Bound (1.3871):** This value represents the maximum expected threshold for the true population mean durability.

#### Final Conclusion
The exact average durability of all print-heads made by the factory is unknown because only 15 sample items were tested. However, based on this test, there is a 99% certainty that the true population mean durability sits somewhere between **1.0902 and 1.3871 million character**.

#### Why the t-distribution Method Was Selected
The t-distribution method was required for two primary reasons:

1.  **Unknown Population Standard Deviation:** The population standard deviation is completely unknown, meaning the sample standard deviation must be used as an estimate instead.
2.  **Small Dataset:** The test only utilized 15 print-heads, which is considered a small sample size (under 30).

Because the population standard deviation is unknown and the test only looked at a small sample size of 15 items, the standard normal (Z) distribution cannot be used. The t-distribution method is necessary because it widens the estimated range slightly to safely account for this missing information.


**b. Build 99% Confidence Interval Using Known Population Standard Deviation** 

If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

In [22]:
## given that standard deviation is 0.2 
sigma=0.2

In [23]:
# Standard Error, SE=sigma/sqrt(n)
SE_z = sigma/ np.sqrt(n)
SE_z

np.float64(0.051639777949432225)

In [24]:
## 99% confidence interval -- Standard Normal (Z) distribution
co_in_z=stats.norm.interval(0.99, loc=x_bar, scale=SE_z) ### (confidence level, mean, standard error)
co_in_z

(np.float64(1.1056514133957607), np.float64(1.3716819199375725))

### Final Result and Method Rationale

#### Lower and Upper Boundary Results
The 99% confidence interval provides an estimated range from **1.1057 to 1.3717 million characters**:

*   **Lower Bound (1.1057):** This value represents the minimum expected threshold for the true population mean durability when the population standard deviation is known.
*   **Upper Bound (1.3717):** This value represents the maximum expected threshold for the true population mean durability when the population standard deviation is known.

#### Final Conclusion
The exact average durability of all print-heads made by the factory is unknown because only 15 sample items were tested. However, based on this test and the known factory settings, there is a 99% certainty that the true population mean durability sits somewhere between **1.1057 and 1.3717 million characters**.

#### Why the Z-distribution Method Was Selected
The standard normal (Z) distribution method was required for this task because:

1.  **Known Population Standard Deviation:** The population standard deviation (sigma = 0.2) is explicitly provided.
2.  **Assumed Normality:** When the population standard deviation is known, the Z-distribution can be used for a small sample size, assuming the underlying population data follows a normal distribution.

Because the population standard deviation is known, it removes the extra uncertainty faced in Task A. This allows the use of the Z-distribution instead of the t-distribution.
